Compose command doc (#12433)
docker/docker.github.io: compose/cli-command-compatibility.md

---
description: Docker Compose command compatibility with docker-compose
keywords: documentation, docs, docker, compose, containers
title: Docker Compose command compatibility with docker-compose
---

In the current state, the compose command in the Docker CLI supports most of the docker-compose commands and flags. It is expected to be a drop in replacement for docker-compose. However, there are a few remaining flags that have not been implemented yet. We are prioritizing the implementation of those based on usage metrics and user feedback.

> If you see some compose functionality that is not available in the compose command, please [let us know](https://github.com/docker/compose-cli/issues){:target="_blank" rel="noopener" class="_"} so we can prioritize it.

Author: Already rephrased after Usha's comments.

## Commands or flags not yet implemented

> The following sub commands are not implemented nor priroitized today. We may add them to our backlog based on user feedback.

Reviewer: The following commands have not been implemented yet, and maybe implemented at a later time.  Please let us know if these commands are a higher priority for your usecases

* `compose build --memory`
* `compose build --no-cache`
* `compose config --no-interpolate`
* `compose config --services`
* `compose config --volumes`
* `compose config --hash`
* `compose events`
* `compose images`
* `compose port`
* `compose pull --ignore-pull-failures`
* `compose push --ignore-push-failures`
* `compose run --service-ports`

## Flags that will not be implemented

Some flags are already deprecated in docker-compose or are not relevant in the go implementation of the compose command in Docker CLI.

> Here is a list of flags that we don't plan to implement in the compose command:

Reviewer: Some of the flags below have been deprecated in docker-compose, or are not relevant for the Compose in Docker CLI.  The list below includes the flags that we are not planning to support in Compose in Docker CLI

> ---

Reviewer 2: This page appears as an orphan page - https://deploy-preview-12433--docsdocker.netlify.app/compose/cli-command-compatibility/. It must be linked from the toc.

Author: I thought linking it from the compose tech preview page was OK. Is it mandatory to make it appear in the TOC. (fine for me if we prefer to link everything in TOC)

Reviewer 2: Yes, we should ideally link it from the toc. If users land on this page, there's no context if they want to go back to a previous page, or select a different topic.

Author: Added `Compose Tech preview compatibility` section just next to `Compose CLI Tech Preview`

Reviewer 2: Yea, agree. We can add that as a subsection.

description: Docker Compose command compatibility with docker-compose
keywords: documentation, docs, docker, compose, containers
title: Docker Compose command compatibility with docker-compose
---

> In the current state, the compose command in the Docker CLI supports most of the docker-compose commands and flags. It is expected to be a drop in replacement for docker-compose. However, there are a few remaining flags that have not been implemented yet. We are prioritizing the implementation of those based on usage metrics and user feedback.

Reviewer 2: ```suggestion
The `compose` command in the Docker CLI supports most of the `docker-compose` commands and flags. It is expected to be a drop-in replacement for `docker-compose`. However, there are a few remaining flags that have not been implemented yet. We are prioritizing the implementation of those based on usage metrics and user feedback.
```

> You can follow progress on the implementation of the remaining commands and flags [here](https://github.com/docker/compose-cli/issues/1283){:target="_blank" rel="noopener" class="_"}.

Reviewer 2: ```suggestion
You can follow progress on the implementation of the remaining commands and flags in the  [Compose-CLI](https://github.com/docker/compose-cli/issues/1283){:target="_blank" rel="noopener" class="_"} GitHub repository.
```

> If you see some compose functionality that is not available in the compose command, please [let us know](https://github.com/docker/compose-cli/issues){:target="_blank" rel="noopener" class="_"} so we can prioritize it.

Reviewer 2: ```suggestion
If you see some Compose functionality that is not available in the `compose` command, create an issue in the [Compose-CLI](https://github.com/docker/compose-cli/issues){:target="_blank" rel="noopener" class="_"} GitHub repository so we can prioritize it.
```

## Commands or flags not yet implemented

The following commands have not been implemented yet, and maybe implemented at a later time.

> Please let us know if these commands are a higher priority for your usecases.

Reviewer 2: ```suggestion
Let us know if these commands are a higher priority for your usecases.
```

* `compose build --memory`
* `compose build --no-cache`
* `compose config --no-interpolate`
* `compose config --services`
* `compose config --volumes`
* `compose config --hash`
* `compose events`
* `compose images`
* `compose port`
* `compose pull --ignore-pull-failures`
* `compose push --ignore-push-failures`
* `compose run --service-ports`

## Flags that will not be implemented

> The list below includes the flags that we are not planning to support in Compose in Docker CLI,

Reviewer 2: ```suggestion
The list below includes the flags that we are not planning to support in Compose in the Docker CLI,
```

> either because they are already deprecated in docker-compose, or because they are not relevant for the Compose in Docker CLI.

Reviewer 2: ```suggestion
either because they are already deprecated in `docker-compose`, or because they are not relevant for Compose in the Docker CLI.
```

> * `compose build --compress` Not relevant as commpose command is using buildkit by default.

Reviewer 2: ```suggestion
* `compose build --compress` Not relevant as the 'compose' command uses buildkit by default.
```

Reviewer 2: Typo: commpose. Could you also fix the following commands based on the suggestion above?

* `compose build --force-rm` Not relevant as commpose command is using buildkit by default.
* `compose build --no-rm` Not relevant as commpose command is using buildkit by default.
* `compose build --parallel` Not relevant as commpose command is using buildkit by default.

> * `compose ps --filter KEY-VALUE` Not relevant as cumbersome usage with the service command and not documented properly in docker-compose.

Reviewer 2: ```suggestion
* `compose ps --filter KEY-VALUE` Not relevant due to its complicated usage with the `service` command and also because it is not documented properly in `docker-compose`.
```

Reviewer 2: On a second thought, wondering if we should remove 'not documented properly' as a reason for deprecating a command?
